# Introdução 
A partir de uma pesquisa do BusinessFinancing.co.uk sobre os empreendimentos mais antigos para diversos países, iremos percorrer os dados para entender as caracteristicas principais desses empreendimentos. 

# 1. Os empreendimentos mais antigos do mundo:

Os datasets utilizados estão localizados na pasta <code>datasets</code> e são divididos da seguinte maneira:
<h3 id="countries"><code>business</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">coluna</th>
<th>tipo</th>
<th>descricao</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>business</code></td>
<td>string</td>
<td>Nome do negócio.</td>
</tr>
<tr>
<td style="text-align:left;"><code>year_founded</code></td>
<td>int</td>
<td>Ano de fundação.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>string</td>
<td>Codigo para a categoria do negócio.</td>
</tr>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>string</td>
<td>Código do Pais (formato ISO 3166-1 3-letter).</td>
</tr>
</tbody>
</table>
<h3 id="countries"><code>countries</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">coluna</th>
<th>tipo</th>
<th>descricao</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>country_code</code></td>
<td>string</td>
<td>Código do País (formato ISO 3166-1 3-letter).</td>
</tr>
<tr>
<td style="text-align:left;"><code>country</code></td>
<td>string</td>
<td>Nome do País.</td>
</tr>
<tr>
<td style="text-align:left;"><code>continent</code></td>
<td>string</td>
<td>Nome do continente aonde o país está.</td>
</tr>
</tbody>
</table>
<h3 id="categories"><code>categories</code></h3>
<table>
<thead>
<tr>
<th style="text-align:left;">coluna</th>
<th>tipo</th>
<th>descricao</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;"><code>category_code</code></td>
<td>string</td>
<td>Código para a categoria do negócio.</td>
</tr>
<tr>
<td style="text-align:left;"><code>category</code></td>
<td>string</td>
<td>Descrição da categoria do negócio.</td>
</tr>
</tbody>
</table>

Antes de descobrir quais são os empreendimentos mais antigos do mundo, iremos realizar um "enriquecimento" da informação através do join entre os 3 datasets. 

In [1]:
# Importando o Pandas para a analise de dados 
import pandas as pd

# Carregando os dataset "businesses.csv", "countries.csv" e "categories.csv" para a variavel business, countries e categories, respectivamente
businesses = pd.read_csv('datasets/businesses.csv')
countries = pd.read_csv('datasets/countries.csv')
categories = pd.read_csv('datasets/categories.csv')

In [2]:
# Criando um dataframe único da união entre businesses, countries e categories para enriquecimento do dado e aprimoramento da analise.
businesses_complete = businesses.merge(countries, on='country_code', how='left')
businesses_complete = businesses_complete.merge(categories, on='category_code', how='left')

display(businesses_complete.head())

business  year_founded category_code country_code  \
0                  Hamoud Boualem          1878         CAT11          DZA   
1  Communauté Électrique du Bénin          1968         CAT10          BEN   
2        Botswana Meat Commission          1965          CAT1          BWA   
3                     Air Burkina          1967          CAT2          BFA   
4                         Brarudi          1955          CAT9          BDI   

        country continent                           category  
0       Algeria    Africa                   Food & Beverages  
1         Benin    Africa                             Energy  
2      Botswana    Africa                        Agriculture  
3  Burkina Faso    Africa               Aviation & Transport  
4       Burundi    Africa  Distillers, Vintners, & Breweries

Após a criação de um DataFrame "enriquecido", iremos partir para algumas analises sobre os dados.  

In [3]:
# Ordenando os dados pela coluna year_founded para vermos os negócios mais antigos do dataset
sorted_businesses = businesses_complete.sort_values(by='year_founded', ascending=True)

sorted_businesses.head()# Criando um dataframe 
# Ordenando os dados pela coluna year_founded para vermos os negócios mais antigos do dataset
sorted_businesses = businesses_complete.sort_values(by='year_founded', ascending=True)

display(sorted_businesses[['business', 'year_founded', 'country', 'category']].reset_index(drop=True).head(10))

business  year_founded         country  \
0                          Kongō Gumi           578           Japan   
1         St. Peter Stifts Kulinarium           803         Austria   
2               Staffelter Hof Winery           862         Germany   
3                   Monnaie de Paris            864          France   
4                      The Royal Mint           886  United Kingdom   
5                          Sean's Bar           900         Ireland   
6              Marinelli Bell Foundry          1040           Italy   
7                    Affligem Brewery          1074         Belgium   
8                         Munke Mølle          1135         Denmark   
9  Ma Yu Ching's Bucket Chicken House          1153           China   

                            category  
0                       Construction  
1          Cafés, Restaurants & Bars  
2  Distillers, Vintners, & Breweries  
3         Manufacturing & Production  
4         Manufacturing & Production  
5          Cafés, Restaurants & Bars  
6         Manufacturing & Production  
7  Distillers, Vintners, & Breweries  
8         Manufacturing & Production  
9          Cafés, Restaurants & Bars

A partir do resultado acima, vemos que o Business mais antigo do mundo é o **"Kongō Gumi"**, fundado no ano de 578, da categoria de Construção e localizado no Japão.

# 2. O négocio mais antigo da América do Sul

In [4]:
# Criando um DataFrame filtrado com os negócios mais antigos da América do Sul
south_america = businesses_complete[businesses_complete['continent']=='South America']
south_america_sorted = south_america[['business', 'year_founded', 'country', 'category']].sort_values(by='year_founded', ascending=True)

display(south_america_sorted.head(10))

business  year_founded  \
157               Casa Nacional de Moneda          1565   
155            Casa de Moneda de Colombia          1621   
159                        Hacienda Chuao          1660   
153               Casa da Moeda do Brasil          1694   
154                                 Famae          1811   
151  Bank of the Province of Buenos Aires          1822   
156                             Banks DIH          1840   
152             Banco Nacional de Bolivia          1871   
158                        Cafe Brasilero          1877   

                               country                    category  
157                               Peru           Banking & Finance  
155                           Colombia  Manufacturing & Production  
159  Venezuela, Bolivarian Republic of            Food & Beverages  
153                             Brazil  Manufacturing & Production  
154                              Chile                     Defense  
151                          Argentina           Banking & Finance  
156                             Guyana            Food & Beverages  
152    Bolivia, Plurinational State of           Banking & Finance  
158                            Uruguay   Cafés, Restaurants & Bars

Aqui, descobrimos que o negócio mais antigo da América do Sul é a **"Casa Nacional de Moneda"** fundada em 1565, localizada no Peru e da categoria de négocios de "Bancos e Finanças" 

# 3. O negócio mais antigo para cada Continente
Agora que vimos o empreendimento mais antigo do Mundo e da América do Sul, vamos buscar qual o empreendimento mais antigo para cada continente.

In [5]:
# Criando um DataFrame auxiliar que contém apenas o continente, e o menor ano de fundação para cada continente.
# Utilizaremos esse DataFrame para encontrar quais são os negócios que tem o menor ano de função em cada continente
continent = businesses_complete.pivot_table(index=['continent'], 
                                             values=['year_founded'], 
                                             aggfunc='min')

# Realizando o Join entre o DataFrame "businesses" e o "continent" para capturar os empreendimentos mais antigos para cada DataFrame. 
oldest_per_continent = continent.merge(businesses_complete, how='left', on='year_founded')[['business', 'year_founded','continent', 'country', 'category']]

display(oldest_per_continent)

business  year_founded      continent    country  \
0               Mauritius Post          1772         Africa  Mauritius   
1                   Kongō Gumi           578           Asia      Japan   
2  St. Peter Stifts Kulinarium           803         Europe    Austria   
3  La Casa de Moneda de México          1534  North America     Mexico   
4               Australia Post          1809        Oceania  Australia   
5      Casa Nacional de Moneda          1565  South America       Peru   

                     category  
0              Postal Service  
1                Construction  
2   Cafés, Restaurants & Bars  
3  Manufacturing & Production  
4              Postal Service  
5           Banking & Finance

# 4. Países sem informação dos negócios mais antigos

Nosso DataSet não contém informações sobre negócios em todos os países. Para localizar os países que não possuimos dados, iremos realizar o join do DataFrame <code>countries</code> com o DataFrame <code>businesses</code> para encontrar os países que não temos informações de business 

In [6]:
# Realizando o Join entre countries e businesses
all_countries = countries.merge(businesses, how='left', on='country_code')  

# Filtrando o DataFrame "all_countries" em que a coluna business tem valor "null" para encontrar os países que não tem informação de negócios.
missing_countries = all_countries[all_countries['business'].isna()]

print(f'Total de países sem dados de negócios antigos: {missing_countries.shape[0]}\n')
display(missing_countries[['country', 'continent']])

Total de países sem dados de negócios antigos: 32



country      continent
1                              Angola         Africa
7                 Antigua and Barbuda  North America
18                            Bahamas  North America
48                 Dominican Republic  North America
50                            Ecuador  South America
57                               Fiji        Oceania
59    Micronesia, Federated States of        Oceania
63                              Ghana         Africa
65                             Gambia         Africa
69                            Grenada  North America
79          Iran, Islamic Republic of           Asia
89                         Kyrgyzstan           Asia
91                           Kiribati        Oceania
92              Saint Kitts and Nevis  North America
107                            Monaco         Europe
108              Moldova, Republic of         Europe
110                          Maldives           Asia
112                  Marshall Islands        Oceania
131                             Nauru        Oceania
138                             Palau        Oceania
139                  Papua New Guinea        Oceania
143                          Paraguay  South America
144               Palestine, State of           Asia
153                   Solomon Islands        Oceania
160                          Suriname  South America
170                        Tajikistan           Asia
171                      Turkmenistan           Asia
172                       Timor-Leste           Asia
173                             Tonga        Oceania
177                            Tuvalu        Oceania
185  Saint Vincent and the Grenadines  North America
189                             Samoa        Oceania

Temos um total de 32 países dos quais não temos dados no nosso Dataset. A lista desses países é exibida acima.

# 5. As indústrias (categorias de negócio) com maior quantidade no nosso dataset

In [7]:
# Criando um DataFrame para contar a quantidade de negócios por categoria
count_business_cats = businesses_complete.groupby('category').agg(business_count=('business','count'))

display(count_business_cats.sort_values(by='business_count', ascending=False))

business_count
category                                         
Banking & Finance                              37
Distillers, Vintners, & Breweries              22
Aviation & Transport                           19
Postal Service                                 16
Manufacturing & Production                     15
Media                                           7
Agriculture                                     6
Food & Beverages                                6
Cafés, Restaurants & Bars                       6
Retail                                          4
Energy                                          4
Tourism & Hotels                                4
Defense                                         3
Consumer Goods                                  3
Mining                                          3
Conglomerate                                    3
Construction                                    2
Telecommunications                              2
Medical                                         1

A partir do resultado acima, descobrimos que a categoria com maior quantidade de negócios é a categoria de "Banking & Finance", seguida por "Distillers, Vintners & Breweries" e por "Aviation & TransportCafés, Restaurants & Bars" 

# 6. Categorias e Continentes

Agora, para finalizar, vamos buscar quando foi fundado o negócio mais antigo por continente e categoria

In [8]:
# Agrupando os dados de menor ano de fundação por continente e categoria
oldest_by_continent_category = businesses_complete.groupby(["continent", "category"]).agg({"year_founded":"min"})
oldest_by_continent_category

year_founded
continent     category                                       
Africa        Agriculture                                1947
              Aviation & Transport                       1854
              Banking & Finance                          1892
              Distillers, Vintners, & Breweries          1933
              Energy                                     1968
              Food & Beverages                           1878
              Manufacturing & Production                 1820
              Media                                      1943
              Mining                                     1962
              Postal Service                             1772
Asia          Agriculture                                1930
              Aviation & Transport                       1858
              Banking & Finance                          1830
              Cafés, Restaurants & Bars                  1153
              Conglomerate                               1841
              Construction                                578
              Defense                                    1808
              Distillers, Vintners, & Breweries          1853
              Energy                                     1928
              Food & Beverages                           1820
              Manufacturing & Production                 1736
              Media                                      1931
              Mining                                     1913
              Postal Service                             1800
              Retail                                     1883
              Telecommunications                         1885
              Tourism & Hotels                           1584
Europe        Agriculture                                1218
              Banking & Finance                          1606
              Cafés, Restaurants & Bars                   803
              Consumer Goods                             1649
              Defense                                    1878
              Distillers, Vintners, & Breweries           862
              Manufacturing & Production                  864
              Media                                      1999
              Medical                                    1422
              Mining                                     1248
              Postal Service                             1520
              Telecommunications                         1912
              Tourism & Hotels                           1230
North America Agriculture                                1638
              Aviation & Transport                       1870
              Banking & Finance                          1891
              Distillers, Vintners, & Breweries          1703
              Food & Beverages                           1920
              Manufacturing & Production                 1534
              Media                                      1909
              Retail                                     1670
              Tourism & Hotels                           1770
Oceania       Banking & Finance                          1861
              Postal Service                             1809
South America Banking & Finance                          1565
              Cafés, Restaurants & Bars                  1877
              Defense                                    1811
              Food & Beverages                           1660
              Manufacturing & Production                 1621